nixos/modules/programs/direnv.nix

```diff
@@ -0,0 +1,88 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}: let
+  cfg = config.programs.direnv;
+in {
+  options.programs.direnv = {
+    enable = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to enable direnv integration. Takes care of both installation and
+        setting up the sourcing of the shell. Additionally enables nix-direnv
+        integration. Note that you need to logout and login for this change to apply.
+      '';
+    };
+    direnvrcExtra = lib.mkOption {
+      type = lib.types.lines;
+      default = "";
+      example = ''
+        use_zig() {
+          zig_inst -q $1
+          ZIG_DIR="''${HOME}/.local/share/zig/$1"
+          ZIG_UPDATE="''${ZIG_DIR}/zig_update"
+          if [ ! -f "''${ZIG_UPDATE}" ]; then
+            echo -e "#!/bin/bash\nzig_inst -q -f $1\n" > "''${ZIG_UPDATE}"
+           chmod a+x "''${ZIG_UPDATE}"
+          fi
+          PATH_add "''${ZIG_DIR}"
+          PATH_add "./zig-cache/bin"
+          PATH_add "./zig-out/bin"
+          echo "Using zig $1"
+        }
+      '';
+      description = ''
+        Extra lines to append to the sourced direnvrc
+      '';
+    };
+    silent = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = ''
+        Whether to hide direnv logging
+      '';
+    };
+    persistDerivations = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      example = false;
+      description = ''
+        Whether to set keep-derivations and keep-outputs to true
+        to prevent shells from getting garbage collected
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    environment = {
+      systemPackages = [pkgs.direnv];
+      variables = {
+        DIRENV_CONFIG = "/etc/direnv";
+        DIRENV_LOG_FORMAT = lib.mkIf cfg.silent "";
+      };
+      etc."direnv/direnvrc".text = ''
+        source ${pkgs.nix-direnv}/share/nix-direnv/direnvrc
+        ${cfg.direnvrcExtra}
+      '';
+    };
+    programs = {
+      zsh.interactiveShellInit = ''
+        eval "$(direnv hook zsh)"
+      '';
+      bash.interactiveShellInit = ''
+        eval "$(direnv hook bash)"
+      '';
+      fish.interactiveShellInit = ''
```

Comment:
What if there is already such an attribute with a custom value? Is this going to override it ?

Reply:
It's of type "list of lines" so it'll be merged

Reply:
Cool ! Looking forward for this thing!